etna/records/templatetags/records_tags.py

```diff
@@ -21,33 +23,60 @@ def record_url(
     Handling of Iaid as priority to allow Iaid in disambiguation pages when
     returning more than one record
     """
-    if iaid := record.get("iaid"):
+    try:
+        # Use Record property if available
+        iaid = record.iaid
+    except AttributeError:
+        # 'record' is likely just a dict
+        iaid = record.get("iaid")
+    except (ValueExtractionError, ValueError):
+        iaid = None
+
+    try:
+        # Use Record property if available
+        ref = record.reference_number
+    except AttributeError:
+        # 'record' is likely just a dict
+        ref = record.get("reference_number", record.get("referenceNumber"))
+    except ValueExtractionError:
+        ref = None
+
+    try:
+        # Use Record property if available
+        url = record.url
+    except AttributeError:
+        # 'record' is likely just a dict
+        url = record.get("sourceUrl")
+    except ValueExtractionError:
+        url = None
```

Comment:
Can we have test cases for the `record_url` template tags?

Reply:
We can, but that's got to be out of scope for this PR, surely?

Reply:
I was looking at test cases within the scope - the new elements introduced when compared with the previous version 
iaid = record.iaid, ref = record.reference_number, url = record.url

Reply:
I haven't really got time to write tests for record_url if they don't already exist. Happy for there to be a ticket created for that though.

Reply:
>if they don't already exist. 

May be misleading as -

>I was looking at test cases within the scope - the new elements introduced when compared with the previous version
iaid = record.iaid, ref = record.reference_number, url = record.url

It would be better in this ticket as its only related to the changes you have made. If you want to take it on a separate ticket,  then please go ahead please address it there cc @sdwilkes  @ACTNA 

This is an important part of this development, if its a separate ticket it would helpfull to see other areas of `record_url` addressed which was previously not.

Reply:
I'll write some tests, but this PR is not the place for them. The new PR should be merged first, then this one rebased on top, so you can see that the changes haven't caused the tests to fail.

Reply:
@TNA-Allan see here for changes needed to get the existing record_url tests to pass (just initialising the `Record` objects differently to match the updated behaviour)... https://github.com/nationalarchives/ds-wagtail/pull/345/commits/753c2de6eff35131774e7ecc097192e405e59717